addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200SwitchProgramService.java

```diff
@@ -0,0 +1,570 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.km200.internal;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+import org.eclipse.smarthome.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The KM200SwitchProgramService representing a switch program service with its all capabilities
+ *
+ * @author Markus Eckhardt - Initial contribution
+ *
+ */
+public class KM200SwitchProgramService {
+    private static final Logger logger = LoggerFactory.getLogger(KM200SwitchProgramService.class);
+
+    protected int maxNbOfSwitchPoints = 8;
```

Comment:
> I guess all the members may be private.

So do I

Reply:
Ok, only the weekdays and times are a kind of constant. (protected final)